v1/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerSchema.java

```diff
@@ -148,6 +150,14 @@ static KeyPart create(String field, boolean desc) {
   @AutoValue
   abstract static class Column implements Serializable {
 
+    private static final Pattern EMBEDDING_VECTOR_PATTERN =
+        Pattern.compile(
+            "^ARRAY<([a-zA-Z0-9]+)>\\(vector_length=>\\d+\\)$", Pattern.CASE_INSENSITIVE);
+
+    private static final Pattern PG_EMBEDDING_VECTOR_PATTERN =
+        Pattern.compile("^(\\D+)\\[\\]\\svector\\slength\\s\\d+$", Pattern.CASE_INSENSITIVE);
```

Comment:
I thought erasable type (vector_length=>) is only for GoogleSQL and in Postgres dialect it is implemented as column option. Is it correct?

Reply:
No. its an erasable type in both googleSQL and Postgres.